clang/test/dpct/syncthreads_in_cond.cu

```diff
@@ -0,0 +1,106 @@
+// UNSUPPORTED: cuda-8.0
+// UNSUPPORTED: v8.0
+// RUN: dpct --format-range=none -out-root %T/syncthreads_in_cond %s --cuda-include-path="%cuda-path/include" --use-experimental-features=nd_range_barrier,logical-group -- -x cuda --cuda-host-only -std=c++14
+// RUN: FileCheck %s --input-file %T/syncthreads_in_cond/syncthreads_in_cond.dp.cpp
+
+#include <cuda_runtime.h>
+
+__device__ void test1(uint64_t *a) {
+  unsigned tid =
+      ((blockIdx.x + (blockIdx.y * gridDim.x)) * (blockDim.x * blockDim.y)) +
+      (threadIdx.x + (threadIdx.y * blockDim.x));
+
+  // check-achor-begin-1
+  __syncthreads();
+  // check-achor-end-1
+  // CHECK: check-achor-begin-1
+  // CHECK-NOT: DPCT1112:{{.*}}: Synchronizing thread blocks inside a conditional branch statement may lead to undefined behavior
```

Comment:
let's not emit L17, instead emit in callee of test1()

Reply:
done